bndtools.core/src/bndtools/central/Central.java

```diff
@@ -89,85 +92,60 @@
 	private static final Supplier<EclipseWorkspaceRepository>	eclipseWorkspaceRepository	= Memoize
 		.supplier(EclipseWorkspaceRepository::new);
 
-	private static Auxiliary									auxiliary;
-
 	static final AtomicBoolean									indexValid					= new AtomicBoolean(false);
 
-	private final BundleContext									bundleContext;
 	private final Map<IJavaProject, Project>					javaProjectToModel			= new HashMap<>();
 	private final List<ModelListener>							listeners					= new CopyOnWriteArrayList<>();
 
 	private RepositoryListenerPluginTracker						repoListenerTracker;
-	private final InternalPluginTracker							internalPlugins;
+	private InternalPluginTracker							internalPlugins;
 
 	@SuppressWarnings("unused")
 	private static WorkspaceRepositoryChangeDetector			workspaceRepositoryChangeDetector;
 
 	private static RepositoriesViewRefresher					repositoriesViewRefresher	= new RepositoriesViewRefresher();
-	private static BundleContext								context;
 	private static ServiceRegistration<Workspace>				workspaceService;
+	private static BundleContext								context;
 
-	static {
-		try {
-			context = FrameworkUtil.getBundle(Central.class)
-				.getBundleContext();
-			Bundle bndlib = FrameworkUtil.getBundle(Workspace.class);
-			auxiliary = new Auxiliary(context, bndlib);
-
-		} catch (Exception e) {
-			// ignore
-		}
-	}
-
-	/**
-	 * WARNING: Do not instantiate this class. It must be public to allow
-	 * instantiation by the Eclipse registry, but it is not intended for direct
-	 * creation by clients. Instead call Central.getInstance().
-	 */
-	@Deprecated
-	public Central() {
-		bundleContext = FrameworkUtil.getBundle(Central.class)
-			.getBundleContext();
-		internalPlugins = new InternalPluginTracker(bundleContext);
-
-	}
-
-	@Override
-	public void start() {
+	@Activate
+	void activate(BundleContext bc) {
+		context = bc;
 		instance = this;
-
-		repoListenerTracker = new RepositoryListenerPluginTracker(bundleContext);
+		repoListenerTracker = new RepositoryListenerPluginTracker(bc);
 		repoListenerTracker.open();
+		internalPlugins = new InternalPluginTracker(bc);
 		internalPlugins.open();
+		// Trigger building of the Workspace object and registering it a service
+
+		promiseFactory().submit(Central::getWorkspace);
```

Comment:
I think that the workspace creation was deferred to avoid prematurely trying to access the Eclipse workspace before it ready. Before my changes, the bndtools.core bundle would fail to activate if it activated too early (before the workbench was running).

With regard to increasing apparently delay: Adding this async call to getWorkspace() here should actually be an improvement in this regard. Previously, the workspace was not initialized until the first time a consumer wished to use it, and it was initialised on whatever thread made that first call to `getWorkspace()`. By submitting this promise, we are pre-emptively triggering the background initialisation of the workspace before any consumer actually needs it - it will be that much closer to ready by the time it is needed.